extension/logstash/lib/logstash/outputs/doris.rb

```diff
@@ -131,9 +132,24 @@ def register
          end
       end
 
+      # Run named Timer as daemon thread
+      @timer = java.util.Timer.new("Doris Output #{self.params['id']}", true)
+
+      @retry_queue = Queue.new
+      retry_thread = Thread.new do
+         while popped = @retry_queue.pop
```

Comment:
Is retry_queue protected by lock to avoid multi thread contention.

Reply:
Yes. Queue in ruby is thread safe.